lib/accounts_session.rb

```diff
@@ -1,8 +1,55 @@
 # frozen_string_literal: true
 
+require "redis"
+require "jwt"
 require "accounts_session/version"
+require "accounts_session/configuration"
+require "accounts_session/session"
+require "accounts_session/stores/redis"
+require "accounts_session/encoders/jwt_token"
+require "accounts_session/decoders/jwt_token"
 
 module AccountsSession
   class Error < StandardError; end
-  # Your code goes here...
+
+  class << self
+    attr_accessor :configuration
+  end
+
+  def self.configuration
+    @configuration ||= Configuration.new
+  end
+
+  def self.configure
+    yield(configuration)
+  end
+
+  def self.reset
+    @configuration = Configuration.new
+  end
+
+  def self.build(session_id:, uuid:, first_name:, last_name:, email:, avatar_url: nil)
+    Session.new(
+      session_id: session_id,
+      uuid: uuid,
+      first_name: first_name,
+      last_name: last_name,
+      email: email,
+      avatar_url: avatar_url
+    )
+  end
+
+  def self.from_token(token)
+    decoded = AccountsSession::Decoders::JwtToken.decode(token)
+    account = decoded.fetch("account")
+
+    build(
+      session_id: decoded.fetch("session_id"),
+      uuid: account.fetch("uuid"),
+      first_name: account.fetch("first_name"),
+      last_name: account.fetch("last_name"),
+      email: account.fetch("email"),
+      avatar_url: account.fetch("avatar_url")
+    )
+  end
```

Comment:
It's tiny, but you can wrap all these method definitions in the `class << self` block and then you can drop the all of the `self.` prefixes

Reply:
+1 I would suggest the same.

Reply:
Right, will do :)